### ISSUE TITLE
Unexpected error location when passing objects/arrays to an argument of type `never`

### ISSUE THREAD

**Issue by mrkev** [microsoft/TypeScript#49449]
# Bug Report

### 🔎 Search Terms

never, argument, unexpected

### 🕗 Version & Regression Information

- This is the behavior in every version I tried, and I reviewed the FAQ for entries about `never`

### ⏯ Playground Link

[Playground link with relevant code](https://www.typescriptlang.org/play?#code/FAEwpgxgNghgTmABAMwK4DsIBcCWB7dRdANwAoAPALiLGLDgEpri8cQBuYYAem8QDEYOKAGdEMMWHIAHSFjAhKwEqQDkIrHBzoA5qobKyAbQBMAGgDMZgCwBdcWPSoAtgCN6R2wa69EAVXQpWWwFFCEoVAQlFVNLGy8ePgBRGTkFakTELIBGRHo4PDhMrJjzKzsDXyzqxAA9eobaxGKaxABBOB0XMHQsRDxkRCwAT1lEVSc3D1tVRBxHPD6JERwddBhXKCQsPERpeBhnMHk4fsGRsYnaelUAOlITC2sAVkruTLbsVBgoRFILsAiCA4ODSLAAAWstwA7LcTEwWhY8nACnAxKQCHkYBAABaIXHCEB5LZHXpvaqlOIVFo1Jp0mnVUgWcgMRAAFVGSAmLnccFm8yIiwcKzWGy2Q12APGgTofPujxM8K4KgA3lRENkAL7klLBeSKZpVIhkNXULXk1qNeoMrIdLqkvoDIac8YqxDqya89iITX8hZLEQi9abba7fZwQ7HehnZ2XGU3eVPV6JD5fH5-AFAkFgyEwuEIo2q9Xmm11UscuM8m5zf3C1bB8U7WNc+Nyh4WRXeLjd4BAA)

### 💻 Code

```typescript
declare function nv(x: never): void;

// Fails as expected:
nv('string')
nv([2,3,4] as number[])


// Unexpected failure:
nv([2,3,4])
// Expected: 
//   1 error
//   nv([2,3,4])
//       ^^^^^^^ 
//       Argument of type 'number[]' is not assignable to parameter of type 'never'.(2345)
//
// Actual (typescirpt@4.7.2):
//   3 errors (on each child element)
//   nv([2,3,4])
//       ^ ^ ^
//       (3x) Type 'number' is not assignable to type 'never'.(2322)

nv({x: 1})
// Expected: 
//   nv({x: 1})
//      ^^^^^^
//       Argument of type '{ x: number; }' is not assignable to parameter of type 'never'.(2345)
//
// Actual (typescirpt@4.7.2):
//   nv({x: 1})
//       ^
//       Type 'number' is not assignable to type 'never'.(2322)
```

### 🙁 Actual behavior

Described in the code comments above.

### 🙂 Expected behavior

Described in the code comments above.
